conda/deprecations.py

```diff
@@ -120,6 +121,35 @@ def inner(*args, **kwargs):
 
         return deprecated_decorator
 
+    def action(
+        self,
+        deprecate_in: str,
+        remove_in: str,
+        action: Action,
+        *,
+        addendum: str | None = None,
+        stack: int = 0,
+    ):
+        class DeprecationMixin:
```

Comment:
I'm not too worried with a slowdown here since it's a deprecation in favor of something else.

### `main` Benchmarks
<img width="755" alt="Screenshot 2023-03-21 at 10 08 14 AM" src="https://user-images.githubusercontent.com/4546435/226649963-ad2e55f6-c6b5-4371-bd84-f90083364f1b.png">

### `kenodegard:argparse-action-deprecation` Benchmarks
<img width="740" alt="Screenshot 2023-03-21 at 10 08 28 AM" src="https://user-images.githubusercontent.com/4546435/226650054-4750ca61-d94a-4d92-a102-8a7142918f7e.png">